packages/smart-forms-renderer/src/stories/itemTypes/AllCalculations.stories.tsx

```diff
@@ -0,0 +1,187 @@
+type Story = StoryObj<typeof meta>;
+
+const booleanTargetCoding = {
+  system: 'http://hl7.org/fhir/administrative-gender',
+  code: 'female',
+  display: 'Female'
+};
+const booleanTargetlinkId = 'gender-controller';
+const booleanTargetlinkIdCalc = 'gender-is-female';
+const qBooleanCalculation = questionnaireFactory(
+  [
+    {
+      linkId: booleanTargetlinkId,
+      text: 'Gender',
+      type: 'choice',
+      repeats: false,
+      answerOption: [
+        { valueCoding: booleanTargetCoding },
+        {
+          valueCoding: {
+            system: 'http://hl7.org/fhir/administrative-gender',
+            code: 'male',
+            display: 'Male'
+          }
+        }
+      ]
+    },
+    {
+      extension: [calculatedExpressionExtFactory("%gender = 'female'")],
+      linkId: booleanTargetlinkIdCalc,
+      text: 'Gender is female?',
+      type: 'boolean',
+      readOnly: true
+    }
+  ],
+  {
+    extension: [
+      variableExtFactory(
+        'gender',
+        `item.where(linkId = '${booleanTargetlinkId}').answer.valueCoding.code`
+      )
+    ]
+  }
+);
+
+export const BooleanCalculation: Story = {
+  args: {
+    questionnaire: qBooleanCalculation
+  },
+  play: async ({ canvasElement }) => {
+    await chooseSelectOption(canvasElement, booleanTargetlinkId, booleanTargetCoding.display);
+
+    await waitFor(async () => {
+      const result = await getAnswers(booleanTargetlinkIdCalc);
+      expect(result).toHaveLength(1);
+      expect(result[0].valueBoolean).toBe(true);
+    });
+  }
+};
+
+const choiceTargetLinkId = 'pain-level';
+const targetChoiceCoding = {
+  system: 'http://terminology.hl7.org/CodeSystem/v2-0532',
+  code: 'Y',
+  display: 'Yes'
+};
+const choiceTargetLinkIdCalc = 'pain-low';
+const qChoiceAnswerOptionCalculation = questionnaireFactory(
+  [
+    {
+      linkId: choiceTargetLinkId,
+      type: 'integer'
+    },
+    {
+      extension: [
+        itemControlExtFactory('radio-button'),
+        calculatedExpressionExtFactory(
+          "iif(%painLevel.empty(), 'Y', iif(%painLevel < 5, 'Y', 'N'))"
+        )
+      ],
+      linkId: choiceTargetLinkIdCalc,
+      text: 'Low pain (Level < 5)',
+      type: 'choice',
+      readOnly: true,
+      answerOption: [
+        {
+          valueCoding: targetChoiceCoding
+        },
+        {
+          valueCoding: {
+            system: 'http://terminology.hl7.org/CodeSystem/v2-0532',
+            code: 'N',
+            display: 'No'
+          }
+        }
+      ]
+    }
+  ],
+  {
+    extension: [
+      variableExtFactory('painLevel', `item.where(linkId = '${choiceTargetLinkId}').answer.value`)
+    ]
+  }
+);
+const choiceTargetNumber = 3;
+
+export const ChoiceAnswerOptionCalculation: Story = {
+  args: {
+    questionnaire: qChoiceAnswerOptionCalculation
+  },
+  play: async ({ canvasElement }) => {
+    await inputText(canvasElement, choiceTargetLinkId, choiceTargetNumber);
+
+    const result = await getAnswers(choiceTargetLinkIdCalc);
```

Comment:
for uniformity let's wrap into await waitFor all these assertions in **all** further tests - it will make clear that we wait results.
Currently it's unclear why in previous case you have waitFor and here - not. We know the answer, because we added 500ms waitfor into inputText, but it's super not obvious